src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaRepository.java

```diff
@@ -99,6 +94,35 @@ public void createTopic(final String topic, final int partitionsNum, final int r
         }
     }
 
+    public boolean topicExists(final String topic) throws NakadiException {
+        return listTopics()
+                .stream()
+                .map(Topic::getName)
+                .anyMatch(t -> t.equals(topic));
+    }
+
+    public boolean areCursorsValid(final String topic, final List<Cursor> cursors) {
+        final List<TopicPartition> partitions = listPartitions(topic);
```

Comment:
it assumes that at this point you already checked that topic exists, and here you only validate cursors.